pkgdb/tests/environment.cc

```diff
@@ -94,14 +102,30 @@ initTestData()
                 { "subtrees", { "legacyPackages" } } } } } } };
   registryWithNixpkgs = RegistryRaw( registryWithNixpkgsJSON );
 
+  lockedRegistryWithNixpkgsJSON
+    = { { "inputs",
+          { { "nixpkgs",
+              { { "from",
+                  { { "type", flox::FLOX_FLAKE_TYPE },
+                    { "owner", "NixOS" },
+                    { "repo", "nixpkgs" },
+                    { "rev", nixpkgsRev },
+                    { "rules", rulesHash },
+                    { "rules-processor", processorHash },
+                    { "narHash", narHashStr } } },
+                { "subtrees", { "legacyPackages" } } } } } } };
+  lockedRegistryWithNixpkgs = RegistryRaw( lockedRegistryWithNixpkgsJSON );
+
   inputWithNixpkgsJSON = { "input",
                            { { "fingerprint", floxNixpkgsFingerprintStr },
-                             { "url", nixpkgsRef },
+                             { "url", ref.to_string() },
                              { "attrs",
-                               { { "owner", "NixOS" },
+                               { { "type", flox::FLOX_FLAKE_TYPE },
+                                 { "owner", "NixOS" },
                                  { "repo", "nixpkgs" },
                                  { "rev", nixpkgsRev },
-                                 { "type", flox::FLOX_FLAKE_TYPE },
+                                 { "rules", rulesHash },
+                                 { "rules-processor", processorHash },
                                  { "narHash", narHashStr } } } } };
```

Comment:
in the resulting lockfile, all flakerefs (outside of `.manifest`) should be locked.